ssl/ssl_lib.c

```diff
@@ -965,11 +965,21 @@ int SSL_set_trust(SSL *s, int trust)
 
 int SSL_set1_host(SSL *s, const char *hostname)
 {
+    /* If a hostname is provided and parses as an IP address,
+     * treat it as such. */
+    if (hostname && X509_VERIFY_PARAM_set1_ip_asc(s->param, hostname) == 1)
```

Comment:
Will do this the right thing if hostname is not NULL, but empty?

Reply:
Yes. It's just that there's no `NULL` check on `X509_VERIFY_PARAM_set1_ip_asc()`.